tools/test_bindings.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Script used to check if bindgen-generated code creates structs that differ from previously created
+onces.
+
+The script uses `pahole` (man 1 pahole) to gather debug information from two firecracker binaries
+(script's arguments). It parses pahole output and gathers struct information in a dictionary of the
+form:
+
+```
+{
+    "struct_name": {"size": size_in_bytes, "alignment": alignment_in_bytes},
+    ...
+}
+```
+
+It also, filters structure names using the "bindings" filter for keeping only bindgen related
+structs.
+
+*NOTE*: this assumes that all bindgen-related structs live under a crate or module name with
+"bindings" in it. At the moment, this is true.
+
+It then iterates through the structs of the firecracker binary built from the older version and
+checks if there are mismatches with the struct info from the second binary (newer version)
+
+### Usage
+
+1. Create the two binaries
+
+```
+# First create the binary with existing bindings
+$ git checkout main
+$ ./tools/devtool build
+$ cp ./build/cargo_target/x86_64-unknown-linux-musl/debug/firecracker firecracker_old
+
+# Second create the binary with new bindings
+$ git checkout new_bindings
+$ ./tools/devtool build
+$ cp ./build/cargo_target/x86_64-unknown-linux-musl/debug/firecracker firecracker_new
+
+# Run the script
+$ python3 ./tools/test_bindings.py firecracker_old firecracker_new
+```
+"""
+
+import sys
```

Comment:
nit: you may want to run it through black / isort / flake8

Reply:
hmmm, the pylint test from the CI doesn't complain. Shouldn't it?

Reply:
Maybe it should, but I think we only do checks within tests due to this: https://github.com/firecracker-microvm/firecracker/blob/main/tools/devtool#L1270

Reply:
Ok, passed it through `black`. Thanks for the suggestion.